cmd/corectl/main.go

```diff
@@ -298,6 +300,72 @@ func reset(client *rpc.Client, args []string) {
 	}
 }
 
+func grant(client *rpc.Client, args []string) {
+	editAuthz(client, args, "grant")
+}
+
+func revoke(client *rpc.Client, args []string) {
+	editAuthz(client, args, "revoke")
+}
+
+type grantReq struct {
+	Policy    string
+	GrantType string      `json:"grant_type"`
+	GrantData interface{} `json:"grant_data"`
+}
+
+func editAuthz(client *rpc.Client, args []string, action string) {
+	usage := "usage: corectl " + action + " [-cn name] [-ou name] [-t token] [policy]"
+	var flags flag.FlagSet
+	flagT := flags.String("t", "", action+" `access-token` access")
+	flagCN := flags.String("cn", "", action+" X.509 `Common Name` access")
+	flagOU := flags.String("ou", "", action+" X.509 `Organizational Unit` access")
+
+	flags.Usage = func() {
+		fmt.Println(usage)
+		flags.PrintDefaults()
+		os.Exit(1)
+	}
+	flags.Parse(args)
+	args = flags.Args()
+	if len(args) < 2 {
+		fatalln(usage)
+	}
+
+	req := grantReq{Policy: args[0]}
+	ok := true
+	if *flagT != "" {
+		req.GrantType = "access_token"
+		req.GrantData = map[string]interface{}{"id": *flagT}
+		ok = ok && editAuthzEntry(client, action, req)
+	}
+	if *flagCN != "" {
+		req.GrantType = "x509"
+		req.GrantData = map[string]interface{}{"subject": map[string]string{"CN": *flagCN}}
+		ok = ok && editAuthzEntry(client, action, req)
+	}
+	if *flagOU != "" {
+		req.GrantType = "x509"
+		req.GrantData = map[string]interface{}{"subject": map[string]string{"OU": *flagOU}}
+		ok = ok && editAuthzEntry(client, action, req)
+	}
+	if !ok {
```

Comment:
So, it's possible that some of these grants were created, right? Should we change this to reflect the possibility that some grants were successfully created, or do we just assume that people will retry (idempotently)?